text/0000-orderer-snapshot.md

```diff
@@ -0,0 +1,263 @@
+
+To check the status of the "join by snapshot" operation use the following REST API:
+
+    GET /v1/mgmt/channels/{channelID}
+
+ or the following CLI command:
+
+    $ osnadmin channel list -c <channelID>
+
+## Guide: Prune block ledger of an existing channel
+[guide-prune-block-ledger-of-an-existing-channel]: #guide-prune-block-ledger-of-an-existing-channel
+
+To prune an existing channel ledger do the following:
+ - create a new snapshot at the desired block number,
+ - remove the OSN from the channel,
+ - join the channel from the snapshot.
+
+This approach has some drawbacks, see [Drawbacks section](#drawbacks).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes to the existing API
+
+This RFC renames the existing **channel participation API** to **channel management API** and introduces three new logical operations (see [Guide-level explanation](#guide-level-explanation)) at this API while keeping the existing channel participation operations, namely:
+- Join this OSN to a new or an existing channel from join-block,
+- Remove this OSN from an existing channel,
+- List the channels this OSN is a member of.
+
+Technically we achieve the above by following these steps:
+1. Deprecate the existing REST API endpoint **/v1/participation/channels** and discourage its use with the plan to drop it completely in versions >= 2.5
+2. Introduce new REST API endpoint **/v1/mgmt/channels** and keep all existing functionality of the channel participation API except "join by join-block", that is keep "remove channel" and "list channel(s)" APIs.
+3. Rename "Channel participation API Configuration" section in the OSN YAML config file to "Channel management API Configuration", rename accordingly all the config settings
+4. Add two new snapshot operations to the REST API definitions of **/v1/mgmt/channels** REST API
+5. Add support of two new snapshot operations to the CLI tool *osnadmin*, specifically to the command *osnadmin channel* and the new command *osnadmin snapshot*
+6. Deprecate *osnadmin channel join* command. This will be the only command using deprecated **/v1/participation/channels**, all the rest of the commands including inherited *osnadmin channel remove* and *osnadmin channel list* must use internally new **/v1/mgmt/channels** endpoint.
+
+## Snapshot content
+
+We propose the OSN snapshot to follow the same logical structure as peer snapshot, that is to comprise a directory with a number of files in it, one of the files being a metadata JSON file with the checksums of other files and other metadata. But here the similarity ends, as the content of the files will be drastically different.
+Here is the content of the OSN snapshot:
+1. Snapshot version - to allow for the future extensions. Current version is 1.
+2. Channel ID.
+3. Last config block for the channel.
+4. lastBlockNum - the number of the last pruned block. This can be both behind the last config block and ahead of it.
```

Comment:
A bit of a nit, but rather than "both .. behind .. and ahead" I believe the intent is "either ... behind ... or ahead".

Reply:
Done, here:
e93aada9af8665466cc0b5777d4ee62f19661e65